griptape/tools/shopify_client/tool.py

```diff
@@ -0,0 +1,66 @@
+from textwrap import dedent
+from typing import Optional
+from urllib.parse import urljoin
+import schema
+from schema import Schema, Literal
+from attr import define, field
+from griptape.tools import BaseTool
+from griptape.utils.decorators import activity
+from griptape.artifacts import BaseArtifact, TextArtifact, ErrorArtifact
+
+
+@define
+class ShopifyClient(BaseTool):
+    storename: str = field(kw_only=True)
+    storefront_access_token: str = field(kw_only=True)
+    shopify_endpoint: str = field(default="myshopify.com/api/2023-07/graphql.json")
+
+    @activity(
+        config={
+            "description": "Can be used to search for shopify products",
+            "schema": Schema(
+                {
+                    Literal("first", description="How many products to retreive."): str,
```

Comment:
Let's rename it to something more descriptive for the LLM. May be `product_count`?

Reply:
*retrieve

Reply:
I really need to review my I's and E's!